mod-source-record-storage-server/src/main/java/org/folio/services/UpdateRecordEventHandler.java

```diff
@@ -0,0 +1,61 @@
+package org.folio.services;
+
+import io.vertx.core.Future;
+import io.vertx.core.json.Json;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.UUID;
+import org.apache.commons.lang.StringUtils;
+import org.folio.processing.events.utils.ZIPArchiver;
+import org.folio.rest.jaxrs.model.ParsedRecordDto;
+import org.folio.rest.jaxrs.model.Record;
+import org.folio.rest.tools.utils.ObjectMapperTool;
+import org.folio.rest.util.OkapiConnectionParams;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import static org.folio.services.util.EventHandlingUtil.sendEventWithPayload;
+
+@Component
+public class UpdateRecordEventHandler {
+
+  private static final Logger LOG = LoggerFactory.getLogger(UpdateRecordEventHandler.class);
+
+  private static final String QM_SRS_MARC_BIB_RECORD_UPDATED_EVENT_TYPE = "QM_SRS_MARC_BIB_RECORD_UPDATED";
+
+  @Autowired
+  private RecordService recordService;
+
+  /**
+   * Handles QM_RECORD_UPDATED event and sends QM_SRS_MARC_BIB_RECORD_UPDATED as a result of the update
+   *
+   * @param eventContent event content
+   * @param params       OkapiConnectionParams
+   * @return future with true if succeeded
+   */
+  public Future<Boolean> handle(String eventContent, OkapiConnectionParams params) {
+    try {
+      HashMap<String, String> eventPayload = ObjectMapperTool.getMapper().readValue(ZIPArchiver.unzip(eventContent), HashMap.class);
```

Comment:
I think it will be better to handle here exception from unzip()-method with the error message "Can`t unzip eventContent".
Moreover, let`s change IOException in 56 line to the common Exception in case of other errors